launcher/resources/git-shim.py

```diff
@@ -59,10 +48,19 @@ def handle_intrinsic(root):
 cmd = json.dumps({ "workingDir": relative_dir, "cmd": ["git"] + sys.argv[1:] })
 s.send(cmd.encode('utf-8') + '\n')
 
-reply = receive_all(s).decode("utf-8")
-s.close()
+socket_file = s.makefile()
+
+for line in socket_file.readlines():
+    response = json.loads(line)
+    if len(response) == 2:
+        if response[0] == 0:
```

Comment:
This encoding isn't obvious, and probably relies on internals of upickle, so please add a line to explain what 0 and 1 stand for.

Reply:
will do